code/modules/mining/lavaland/weather_types.dm

```diff
@@ -0,0 +1,26 @@
+/datum/weather/floor_is_lava
+	name = "floor is lava"
+	start_up_time = 30 //3 seconds
+	start_up_message = "The ground begins to bubble."
+	duration = 60 //1 minute
+	duration_message = "The floor is lava!"
+	wind_down = 30// 3 seconds
+	wind_down_message = "The ground begins to cool."
+
+	target_z = 1
+	exclude_walls = TRUE
+	area_type = /area
```

Comment:
What I really need to do is update all the areas to /area/station because that would solve a lot of other problems as well.

Station plopping down into a lake of fire sounds amusing in the mean time though